src/main/java/online/talkandtravel/service/impl/OnlineServiceImpl.java

```diff
@@ -0,0 +1,140 @@
+        }
+
+        notifySubscribedUsers(user.getId(), isOnline);
+    }
+
+    @Override
+    public void notifyUserOnlineStatusUpdated(Long userId, Boolean isOnline) {
+        notifySubscribedUsers(userId, isOnline);
+    }
+
+    /**
+     * Provides an initial snapshot of all user's online statuses
+     *
+     * @return a map where the key is the user ID and the value is a boolean indicating
+     * whether the user is online (true) or offline (false)
+     */
+    @Override
+    public Map<Long, Boolean> getAllUsersOnlineStatuses() {
+        log.info("getAllUsersOnlineStatuses");
+
+        Set<String> keys = redisTemplate.keys(USER_STATUS_KEY_PATTERN);
+        List<Long> userIdList = getUserIdFromKeys(keys);
+        List<String> values = redisTemplate.opsForValue().multiGet(keys);
+
+        log.info("keys: {}, values: {}", keys, values);
+        return mapKeysAndValuesToMap(userIdList, values);
+    }
+
+    /**
+     * Provides a current online status of a specific user
+     *
+     * @param userId user id
+     * @return a map where the key is the user ID and the value is a boolean indicating
+     * whether the user is online (true) or offline (false)
+     */
+    @Override
+    public Boolean getUserOnlineStatusById(Long userId) {
+        String key = getRedisKey(userId);
+        String value = redisTemplate.opsForValue().get(key);
+        Boolean isOnline = Boolean.valueOf(value);
+
+        log.info("getUserOnlineStatusById: {}, isOnline={}", userId, isOnline);
+        return isOnline;
+    }
+
+    /**
+     * Retrieves the current online statuses of a specified list of users.
+     *
+     * @param userIds the list of user IDs for which to fetch online statuses
+     * @return a map where the key is the user ID and the value is a boolean indicating
+     * whether the user is online (true) or offline (false)
+     */
+    @Override
+    public Map<Long, Boolean> getAllUsersOnlineStatusesForUsersList(List<Long> userIds) {
+        List<String> keys = getRedisKeys(userIds);
+        List<String> values = redisTemplate.opsForValue().multiGet(keys);
+
+        log.info("getAllUsersOnlineStatusesForUsersList {}", values);
+        return  mapKeysAndValuesToMap(userIds, values);
+    }
+
+    private void notifySubscribedUsers(Long userId, Boolean isOnline) {
+        UserOnlineStatusDto dto = new UserOnlineStatusDto(userId, isOnline);
+        messagingTemplate.convertAndSend(USERS_ONLINE_STATUS_ENDPOINT, dto);
+    }
+
+    private Map<Long, Boolean> mapKeysAndValuesToMap(List<Long> userIdList, List<String> values) {
+        Map<Long, Boolean> onlineStatuses = new HashMap<>(userIdList.size());
+
+        int i = 0;
+        for (Long userId : userIdList) {
+            onlineStatuses.put(userId, Boolean.valueOf(values.get(i++)));
+        }
+        log.info("onlineStatuses map: {}", onlineStatuses);
+        return onlineStatuses;
+    }
+
+    private List<Long> getUserIdFromKeys(Set<String> keys) {
```

Comment:
why are not all methods below in 'RedisUtil'

Reply:
or move that single method from RedisUtil to OnlineService. (or move all these methods to RedisUtil)